src/collection.ts

```diff
@@ -459,26 +459,29 @@ export class Collection<TSchema extends Document = Document> {
    * @param options - Optional settings for the command
    * @param callback - An optional callback, a Promise will be returned if none is provided
    */
-  replaceOne(filter: Filter<TSchema>, replacement: TSchema): Promise<UpdateResult | Document>;
   replaceOne(
     filter: Filter<TSchema>,
-    replacement: TSchema,
+    replacement: OptionalId<TSchema>
```

Comment:
I think we should instead change these to `WithoutId` whether or not you are using ObjectId the _id field is always immutable, so if its ever specified here you get an error from the server. Might as well catch that in TS here!

Reply:
Yeah, fair enough.